nucypher/cli/characters/alice.py

```diff
@@ -165,71 +177,87 @@ def alice(click_config,
         ALICE.log.info('Starting HTTP Character Web Controller')
         return controller.start(http_port=http_port, dry_run=dry_run)
 
+    elif action == "destroy":
+        """Delete all configuration files from the disk"""
+        if dev:
+            message = "'nucypher ursula destroy' cannot be used in --dev mode"
+            raise click.BadOptionUsage(option_name='--dev', message=message)
+        return actions.destroy_configuration(character_config=alice_config, force=force)
+
+    #
+    # Alice API
+    #
+
     elif action == "public-keys":
         response = ALICE.controller.public_keys()
         return response
 
     elif action == "create-policy":
+
+        # Validate
         if not all((bob_verifying_key, bob_encrypting_key, label)):
             raise click.BadArgumentUsage(message="--bob-verifying-key, --bob-encrypting-key, and --label are "
                                                  "required options to create a new policy.")
 
+        # Request
         create_policy_request = {
             'bob_encrypting_key': bob_encrypting_key,
             'bob_verifying_key': bob_verifying_key,
             'label': label,
             'm': m,
             'n': n,
         }
-
         return ALICE.controller.create_policy(request=create_policy_request)
 
     elif action == "derive-policy-pubkey":
+
+        # Validate
         if not label:
-            raise click.BadOptionUsage(option_name='label',
-                                       message="--label is required for deriving a policy encrypting key.")
+            raise click.BadOptionUsage(option_name='label', message="--label is required for deriving a policy encrypting key.")
+
+        # Request
         return ALICE.controller.derive_policy_encrypting_key(label=label)
 
     elif action == "grant":
+
+        # Validate
+        if not all((bob_verifying_key, bob_encrypting_key, label)):
+            raise click.BadArgumentUsage(message="--bob-verifying-key, --bob-encrypting-key, and --label are "
+                                                 "required options to grant (optionally --m, --n, and --expiration).")
+
```

Comment:
The presence of `--value` should be validated in case of decentralized Alice.

Reply:
Yeah, OK.  What can we check here?

Reply:
:heavy_check_mark: